### PR TITLE
Add --validators-graffiti command line option

### PR DESCRIPTION
**Pull request by rolfyone**
Add --validators-graffiti command line option to give graffiti to use during block production.

Addresses #1750 

Signed-off-by: Paul Harris <paul.harris@consensys.net>

### CODE REVIEW

File: teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorOptions.java [hunk 1]
```diff
@@ -64,6 +66,15 @@
       arity = "1")
   private int validatorExternalSignerTimeout = 1000;
 
+  @Option(
+      names = {"--validators-graffiti"},
+      converter = GraffitiConverter.class,
+      paramLabel = "<GRAFFITI STRING>",
+      description =
+          "Graffiti to include during block creation. (gets converted to bytes and padded to Bytes32)",
```
Comment (macfarla) [inline teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorOptions.java]:
```suggestion
          "Graffiti to include during block creation (gets converted to bytes and padded to Bytes32).",
```

File: util/src/main/java/tech/pegasys/teku/util/cli/GraffitiConverter.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  @Override
+  public Bytes32 convert(final String value) {
+    byte[] input = value.getBytes(StandardCharsets.UTF_8);
+    if (input.length > 32) {
+      throw (new CommandLine.TypeConversionException(
+          "'"
+              + value
+              + "' converts to "
+              + input.length
+              + " bytes. Only a maximum of 32 bytes can be used as graffiti."));
```
Comment (macfarla) [inline util/src/main/java/tech/pegasys/teku/util/cli/GraffitiConverter.java]:
```suggestion
              + " bytes. Maximum of 32 bytes can be used as graffiti."));
```

File: util/src/main/java/tech/pegasys/teku/util/cli/GraffitiConverter.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+package tech.pegasys.teku.util.cli;
+
+import java.nio.charset.StandardCharsets;
+import org.apache.tuweni.bytes.Bytes32;
+import picocli.CommandLine;
+
+public class GraffitiConverter implements CommandLine.ITypeConverter<Bytes32> {
+  @Override
+  public Bytes32 convert(final String value) {
```
Comment (macfarla) [inline util/src/main/java/tech/pegasys/teku/util/cli/GraffitiConverter.java:22]: is there a test for this?

File: teku/src/test/java/tech/pegasys/teku/cli/options/ValidatorOptionsTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.cli.AbstractBeaconNodeCommandTest;
+import tech.pegasys.teku.datastructures.util.DataStructureUtil;
+import tech.pegasys.teku.util.config.TekuConfiguration;
+
+public class ValidatorOptionsTest extends AbstractBeaconNodeCommandTest {
+  private final DataStructureUtil dataStructureUtil = new DataStructureUtil();
+  private final Bytes32 graffiti =
+      Bytes32.fromHexString("0x542045204b205500000000000000000000000000000000000000000000000000");
```
Comment (macfarla) [inline teku/src/test/java/tech/pegasys/teku/cli/options/ValidatorOptionsTest.java]: is this the hex of the "T E K U" string that is in the config file? Can the test verify this?

**Review by macfarla — COMMENTED** [review]
couple of nits and a test

File: teku/src/test/java/tech/pegasys/teku/cli/options/ValidatorOptionsTest.java (hunk 4 above)
Comment (rolfyone) [reply teku/src/test/java/tech/pegasys/teku/cli/options/ValidatorOptionsTest.java]: its annoying, without converting the string with the same functionality in the converter, which would just be duplicating the code. I manually verified with a Hex->utf8 page.
The same logic applies for the test cases for the converter itself, its not a trivial function to convert the text to hex (at least that i found)

**Review by macfarla — APPROVED** [review]
LGTM

File: teku/src/test/java/tech/pegasys/teku/cli/options/ValidatorOptionsTest.java (hunk 4 above)
Comment (macfarla) [reply teku/src/test/java/tech/pegasys/teku/cli/options/ValidatorOptionsTest.java]: a comment in the test would make it clearer then

Comment (rolfyone) [reply teku/src/test/java/tech/pegasys/teku/cli/options/ValidatorOptionsTest.java]: moved it to `shouldReadFromConfigurationFile` and added comment

**Pull request merged**